rust/headless-client/src/imp_linux.rs

```diff
@@ -231,13 +240,16 @@ fn system_resolvers(dns_control_method: Option<DnsControlMethod>) -> Result<Vec<
 }
 
 #[derive(Clone)]
-struct CallbackHandler;
+struct CallbackHandler {
+    on_disconnect_tx: mpsc::Sender<()>,
+}
 
 impl Callbacks for CallbackHandler {
     fn on_disconnect(&self, error: &connlib_client_shared::Error) {
         tracing::error!("Disconnected: {error}");
-
-        std::process::exit(1);
+        self.on_disconnect_tx
+            .try_send(())
```

Comment:
Disconnecting via the callback is always an error. I'd say instead of logging the error above, send it through the channel and return it as an error from main to reflect in the exit code that we failed.

Reply:
Yeah that looks better with the exit code and everything. I made it into a String, some of the Error variants are not cloneable I think.